File: fe_backup/src/shared/lib/supabase.ts
import { createClient } from "@supabase/supabase-js";

// Khởi tạo Supabase client cho frontend
// Cần đảm bảo các biến môi trường này đã được thiết lập trong .env.local
const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!;

export const supabase = createClient(supabaseUrl, supabaseKey);

File: backend/app/core/supabase.py
from supabase import Client, create_client
from app.core.config import settings

def get_supabase_client() -> Client:
    """
    Khởi tạo Supabase Client với quyền Admin (Service Role).
    Dùng để thực hiện các tác vụ quản trị như Invite User, Delete User.
    CRITICAL: Không bao giờ expose client này ra Frontend vì nó có full quyền.
    """
    if not settings.SUPABASE_URL or not settings.SUPABASE_SERVICE_ROLE_KEY:
        raise ValueError("Thiếu cấu hình SUPABASE_URL hoặc SUPABASE_SERVICE_ROLE_KEY")

    return create_client(settings.SUPABASE_URL, settings.SUPABASE_SERVICE_ROLE_KEY)

supabase_admin = get_supabase_client()
